module/translation-deepl/src/Infrastructure/Provider/Decorator/TranslationDeeplProviderDecorator.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See license.txt for license details.
+ */
+
+declare(strict_types = 1);
+
+namespace Ergonode\TranslationDeepl\Infrastructure\Provider\Decorator;
+
+use Ergonode\Core\Domain\ValueObject\Language;
+use Ergonode\TranslationDeepl\Infrastructure\Cache\DatabaseTranslationCache;
+use Ergonode\TranslationDeepl\Infrastructure\Provider\TranslationDeeplProvider;
+use Ergonode\TranslationDeepl\Infrastructure\Provider\TranslationDeeplProviderInterface;
+use Ramsey\Uuid\Uuid;
+
+/**
+ */
+class TranslationDeeplProviderDecorator implements TranslationDeeplProviderInterface
```

Comment:
The current name of the class does not tell us about its purpose.
it should be named: CacheTranslationProviderDecorator